### PR TITLE
Add filter iterators for weighted set blueprints.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@arnej27959 : FYI

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+DocumentWeightOrFilterSearchImpl::~DocumentWeightOrFilterSearchImpl() = default;
+
+void
+DocumentWeightOrFilterSearchImpl::doSeek(uint32_t docId)
+{
+    if (_children.get_docid(0) < docId) {
+        _children.seek(0, docId);
+    }
+    uint32_t min_doc_id = _children.get_docid(0);
+    for (unsigned int i = 1; min_doc_id > docId && i < _children.size(); ++i) {
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp:56]: _children.size() returns a value of type size_t, while _children.get_docid() and _children.seek() expects argument of type uint16_t. Perhaps it's better to use uint16_t ?

[the remark this answers is not in the document]
Comment (geirst) [reply searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp:56]: Yes, I agree

[the remark this answers is not in the document]
Comment (toregge) [reply searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp]: Fixed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**